### PR TITLE
Migrate 'LegendProportion' component from makeStyles to styled-components + cleanup

### PR DESCRIPTION
**Pull request by jantolg**
# Description

This PR includes a refactorization of LegendProportion component, all classes from makestyles are been converted to styled components

Shortcut:  [297037](https://app.shortcut.com/cartoteam/story/297037/migrate-away-from-makestyles-i-widgets)

## Type of change

- Refactor

# Acceptance

The code resultant must return same styles component visualization as the same before changes

1. Overview a general visualization from v1.x to v2.x

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

File: packages/react-ui/src/widgets/legend/LegendProportion.js
Comment (vmilan) [inline packages/react-ui/src/widgets/legend/LegendProportion.js:16]: You could use `theme.spacing()` also here.

Take a look at this component to see an example: `packages/react-ui/src/components/molecules/Avatar.js`

**Review by vmilan — COMMENTED** [review]
Missing `changelog` note and there is a visual difference between your branch and master, please take a look.

Screenshot 2023-04-12 at 15 00 37

**Comment by aaranadev** [comment]
I've fixed the visual difference

File: packages/react-ui/src/widgets/legend/LegendProportion.js
Comment (aaranadev) [reply packages/react-ui/src/widgets/legend/LegendProportion.js:57]: Done

**Review by vmilan — APPROVED** [review]
UI LGTM, just adds a note in the changelog please, and wait for @VictorVelarde review before merging.

Thanks!!

**Review by VictorVelarde — APPROVED** [review]
LGTM, please changelog and we're free to land, thx!

**Pull request merged**
